File: tests/python/greedy_algorithms/class_photos/test_class_photos.py
from problems.greedy_algorithms.class_photos.python.class_photos import \
    classPhotos
from tests.python.utility import load_test_cases


def test_class_photos() -> None:
    test_cases = load_test_cases("greedy_algorithms/class_photos.json")

    for idx, test in enumerate(test_cases):
        redShirtHeights: list[int] = test["redShirtHeights"]
        blueShirtHeights: list[int] = test["blueShirtHeights"]
        expected: bool = test["expected"]

        result: bool = classPhotos(redShirtHeights, blueShirtHeights)
        assert (
            result == expected
        ), f"Test case {idx} failed: expected {expected}, got {result}"

File: tests/cpp/greedy_algorithms/test_class_photos.cpp
#include "../../../problems/greedy_algorithms/class_photos/cpp/class_photos.h"
#include "../jsontestbase.h"

#include <vector>

class ClassPhotosTest : public JsonTestBase {
public:
  ClassPhotosTest() {
    json_file_path =
        std::string(TEST_CASES_DIR) + "/greedy_algorithms/class_photos.json";
  }
};

TEST_F(ClassPhotosTest, TestCases) {
  ASSERT_FALSE(test_cases.empty()) << "Test cases are empty or not loaded";

  for (const auto &test : test_cases) {
    auto redShirtHeights = test["redShirtHeights"].get<std::vector<int>>();
    auto blueShirtHeights = test["blueShirtHeights"].get<std::vector<int>>();
    bool expected = test["expected"].get<bool>();

    auto result = classPhotos(redShirtHeights, blueShirtHeights);

    EXPECT_EQ(expected, result) << "Failed for test case: " << test.dump(2);
  }
}
